File: backend/src/entities/album.php
<?php

class Album implements JsonSerializable {
  
  private $id;
  private $uuid;
  private $title;
  private $artist;
  private $coverName;
  private $tracks;


  public function __construct($id, $uuid, $title, $artist, $coverName, $tracks) {
    $this->id = $id;
    $this->uuid = $uuid;
    $this->title = $title;
    $this->artist = $artist;
    $this->coverName = $coverName;
    $this->tracks = $tracks;
  }

  public function jsonSerialize() {
    return [
        'id' => $this->id,
        'uuid' => $this->uuid,
        'title' => $this->title,
        'artist' => $this->artist,
        'coverName' => $this->coverName,
        'tracks' => $this->tracks
    ];
  }

}

?>

File: android/app/src/main/kotlin/pl/kwasow/data/Album.kt
package pl.kwasow.data

import kotlinx.serialization.Serializable

@Serializable
data class Album(
    val id: Int,
    val uuid: String,
    val title: String,
    val artist: String,
    val coverName: String,
    val tracks: List<AudioTrack>,
)
